File: LeetCode/2225/solution.cpp
// 2225. Find Players With Zero or One Losses
class Solution {
  public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
      unordered_map<int,int> mp;

      for(auto match: matches){
        mp[match[0]] += 0;
        mp[match[1]] += 1;
      }

      vector<vector<int>>ans(2);
      for(auto a: mp){
        if(a.second==0) ans[0].push_back(a.first);
        if(a.second==1) ans[1].push_back(a.first);
      }

      sort(ans[0].begin(), ans[0].end());
      sort(ans[1].begin(), ans[1].end());

      return ans;
    }
};

File: LeetCode/2225/solution.java
// 2225. Find Players With Zero or One Losses
class Solution {
  public List<List<Integer>> findWinners(int[][] matches) {
    Map<Integer, Integer> map= new HashMap<>();

    for(int[] match: matches){
      map.put(match[0], map.getOrDefault(match[0], 0) + 0);
      map.put(match[1], map.getOrDefault(match[1], 0) + 1);
    }

    List<List<Integer>> ans = new ArrayList<>();
    ans.add(new ArrayList<>());
    ans.add(new ArrayList<>());

    for(Map.Entry<Integer, Integer> entry : map.entrySet()){
      if(entry.getValue()==0) ans.get(0).add(entry.getKey());
      if(entry.getValue()==1) ans.get(1).add(entry.getKey());
    }
    
    Collections.sort(ans.get(0));
    Collections.sort(ans.get(1));

    return ans;
  }
}